### PR TITLE
Added more validation to the passwords.

### PR DESCRIPTION
**Pull request by jesuisnicolas**
I added extra validation to the password on signup.
Now passwords validate for:
* 8 characters minimum.
* 1 uppercase character
* 1 lowercase character
* 1 number.
* 1symbol.

I also added a message in the signup form with the requirements that shows what validation passes.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
         "morgan": "^1.10.0",
         "nodemailer": "^6.7.0",
         "nodemon": "^2.0.14",
+        "password-validator": "^5.2.1",
```
Comment (MATTALUI) [inline package-lock.json:29]: The lock has changed, but not the package. This will cause deployment issues.

File: src/static/js/signup.js [hunk 2]
```diff
@@ -1,14 +1,85 @@
...
+
+// When the user clicks on the password field, show the message box
+password.onfocus = function() {
+  document.getElementById("passwordInstructions").style.display = "block";
+}
+
+// When the user clicks outside of the password field, hide the message box
+password.onblur = function() {
+  document.getElementById("passwordInstructions").style.display = "none";
+}
```
Comment (MATTALUI) [inline src/static/js/signup.js:33]: Use event listeners here instead. This is like inline JS, which is bad practice.

File: src/static/js/signup.js [hunk 3]
```diff
@@ -1,14 +1,85 @@
-let confirmPassword = document.getElementById("confirmPassword");
+
```
Comment (MATTALUI) [inline src/static/js/signup.js:1]: If you wrap this all in an IIFE, you will prevent people from messing with this through the browser console.

**Review by MATTALUI — CHANGES_REQUESTED** [review]
We cant merge this until the `package.json` files reflect any new packages otherwise it will break things.

File: package-lock.json (hunk 1 above)
Comment (jesuisnicolas) [reply package-lock.json]: Fixed. I deleted all the references to the password-validator.

File: src/static/js/signup.js (hunk 2 above)
Comment (jesuisnicolas) [reply src/static/js/signup.js]: Done.

File: src/static/js/signup.js (hunk 3 above)
Comment (jesuisnicolas) [reply src/static/js/signup.js:5]: Done.

**Review by MATTALUI — APPROVED** [review]

**Pull request merged**